### ISSUE TITLE
Server side redirects result in a timeout on Vercel

### ISSUE THREAD

**Issue by PupoSDC** [vercel/next.js#32745]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

16.13.1

### What browser are you using?

chromium

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

Server side Redirects don't work when deployed on vercel. Result in 

```
504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
```

### Expected Behavior

redirects should not exceed timeout limits.

### To Reproduce

```ts
// pages/[...redirectAll].tsx
import { FunctionComponent } from "react";
import { GetServerSideProps } from "next";

const RedirectAll: FunctionComponent = () => <></>;

export const getServerSideProps: GetServerSideProps = async () => ({
  redirect: {
    permanent: false,
    destination: "/",
  },
});

export default RedirectAll;
```

deployed here:

https://widu-webapp.vercel.app/potato

results in a 

```
504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
```

works fine locally, and on AWS serverless.

**Comment by balazsorban44** [comment]
Hi, I tried doing the same, but I could not reproduce. Here is my deployment:

https://32745-i1hjnhekj-balazsorban.vercel.app

https://32745-i1hjnhekj-balazsorban.vercel.app/potato

Could you add a reproduction/share more of your code please, so we can investigate? :pray:

My code was:

```js
//pages/index.tsx
export default function Home() {
  return "Home"
}
```

```js
// pages/[...redirectAll].tsx
import { FunctionComponent } from "react"
import { GetServerSideProps } from "next"

const RedirectAll: FunctionComponent = () => <></>

export const getServerSideProps: GetServerSideProps = async () => ({
  redirect: {
    permanent: false,
    destination: "/",
  },
})

export default RedirectAll
```

**Comment by PupoSDC** [comment]
Would it be ok to invite you over a private repo?

**Comment by balazsorban44** [comment]
Thabks, I'll have a look

**Comment by tooppaaa** [comment]
I'm interested by your findings @balazsorban44, thanks !

**Comment by balazsorban44** [comment]
@tooppaaa please give a reproduction as well then. There is a big chance the issue lies in some kind of configuration, as you can see above, I could not reproduce it with the given basic reproduction.

**Comment by PupoSDC** [comment]
I was able to fix this issue.  To be honest, i dont fully understand what happens under the hood, but I have fairly good gut feeling.

Basically, the problem revolves around `next-18next`, combined with the way my app was setup. Basically my app used to look like this:

```ts
// _app.tsx
const App = ({
  Component,
  emotionCache = clientSideEmotionCache,
  pageProps,
  session = null,
}: CustomAppProps) => (
  <StrictMode>
       </BunchOfProviders>
            <PageHeaderThatUsedTranslations />
            <Component {...pageProps} />
        </BunchOfProviders>
  </StrictMode>
);
```

so when we hit a redirect page there seems to be some translation happening. In fact, with this setup, no server side page works at all. I seem to be hitting the same problem as this guy: https://github.com/vercel/next.js/discussions/32686

I tried a lot of intermediate solutions, like adding the translation setup on my redirect page but nothing worked. In the end the only thing that worked was moving away the `<PageHeaderThatUsedTranslations />` to be included in every `Page` so that translations are not needed unless you ask for it.

I was a bit of a painful experience since there are no logs on Vercel side, and this is not reproducible on the client side. My wild guess is that on vercel the `_app.tsx` code gets run before `getServerSideProps` resulting in some translations getting stuck waiting for resources.

thank you for having a look into this!
